script.js

```diff
@@ -1,4 +1,126 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+/* 
+User input = Scissors/Paper/Stone
+Output = Win/Lose
+    1        2       3        1
+Scissors > Paper > Stone > Scissors
+
+0-0 = 0
+0-1 = -1
+0-2 = -2
+
+1-0 = 1
+1-1 = 0
+1-2 = -1
+
+2-0 = 2
+2-1= 1
+2-2 = 0
+
+ */
+// Gobal Variables
+const selection = ["scissors", "paper", "stone"];
+var userWinLoseRecord = 0;
+var aiWinLoseRecord = 0;
+var totalRuns = 0;
+var userName = "";
+var userNameFlag = 0;
+
+var getRandomArray = function (lenOfArray) {
+  // Generate a decimal from 0 through 6, inclusive of 0 and exclusive of
+  var randomDecimal = Math.random() * lenOfArray;
+
+  // Remove the decimal with the floor operation.
+  // This will be an integer from 0 to 5 inclusive.
+  var randomInteger = Math.floor(randomDecimal);
+
+  return randomInteger;
+};
+
+var getAiOutput = function () {
+  var aiOutput = selection[getRandomArray(selection.length)];
+  return aiOutput;
+};
+
+var checkUserInput = function (userSelection) {
+  if (
+    userSelection == "scissors" ||
+    userSelection == "paper" ||
+    userSelection == "stone" ||
+    userSelection == "reversed scissors" ||
+    userSelection == "reversed paper" ||
+    userSelection == "reversed stone"
+  ) {
+    var flag = 1;
+  } else {
+    flag = 0;
+  }
+  return flag;
+};
+
+var checkPlayerWin = function (userWinInput, aiLoseInput) {
+  return (
+    (userWinInput == "scissors" && aiLoseInput == "paper") ||
+    (userWinInput == "paper" && aiLoseInput == "stone") ||
+    (userWinInput == "stone" && aiLoseInput == "scissors") ||
+    (userWinInput == "reversed scissors" && aiLoseInput == "stone") ||
+    (userWinInput == "reversed paper" && aiLoseInput == "scissors") ||
+    (userWinInput == "reversed stone" && aiLoseInput == "paper")
+  );
+};
+
+var checkPlayerDraw = function (userDrawInput, aiDrawInput) {
+  return (
+    userDrawInput == aiDrawInput ||
+    (userDrawInput == "reversed scissors" && aiDrawInput == "scissors") ||
+    (userDrawInput == "reversed paper" && aiDrawInput == "paper") ||
+    (userDrawInput == "reversed stone" && aiDrawInput == "stone")
+  );
+};
+
+var main = function (userInput) {
+  var userInputFlag = checkUserInput(userInput);
```

Comment:
Excellent way of validating your inputs! I admit I will have to take this leaf out of your book

Reply:
Thank you for your comments Bryan. I actually got the idea from the reference code provided, Its a great thought on that, and i want to implement into my code so that i can remember in the future,